test/State.hs

```diff
@@ -7,29 +7,36 @@ module State
 
 import qualified Control.Carrier.State.Lazy as LazyStateC
 import qualified Control.Carrier.State.Strict as StrictStateC
-import Control.Effect.State
+import           Control.Effect.State
+import           Control.Exception (SomeException, evaluate, try)
+import           Control.Monad
+import           Control.Monad.IO.Class
 import qualified Control.Monad.Trans.RWS.Lazy as LazyRWST
 import qualified Control.Monad.Trans.RWS.Strict as StrictRWST
 import qualified Control.Monad.Trans.State.Lazy as LazyStateT
 import qualified Control.Monad.Trans.State.Strict as StrictStateT
-import Data.Tuple (swap)
-import Gen
+import           Data.Either
+import           Data.Tuple (swap)
+import           Gen
+import           Hedgehog (assert, eval)
 import qualified Monad
 import qualified MonadFix
-import Test.Tasty
-import Test.Tasty.Hedgehog
+import           Test.Tasty
+import           Test.Tasty.Hedgehog
 
 tests :: TestTree
 tests = testGroup "State"
   [ testGroup "StateC (Lazy)"   $
     [ testMonad
     , testMonadFix
     , testState
+    , testLazy
```

Comment:
To verify this actually works, add a `testStrict` here (and a `testLazy` below) and observe as these operations fail despite operating over the same expression.